### ISSUE TITLE
[Suggestion] Simple Distance Sensor

### ISSUE THREAD

**Issue by Peekofwar** [Seniorendi/AdvancedPeripherals#320]
### Describe your idea

Would be nice to have some sort of simple distance sensor that would output the distance between it and a block in front of it in a straight line.

Possible deployments:
* Detect where a Create Mod elevator cabin stops in a shaft to confirm/calibrate the positioning of the lift
* Detect an entity or intruder in an area by checking for a change in distance (could be done with tripwire hooks, but is less clean and technically takes up more space)

Possible methods/functions exposed to the programmer:
* `setLaserVisible(boolean)` enables a visible laser (default off)
* `getLaserVisible()` returns `true` if the laser is set to visible, else `false`
* `setDetectTransparent(boolean)` would make it detect transparent blocks (default off)
* `getDetectTransparent()` returns `true` if it is set to detect transparent blocks, else `false`
* `setHeightMode(value)` sets a `value` that determines which height the sensor's laser is emitted from. 
  * This could be a predetermined number of modes like `1` is low, `2` is center, and `3` is high,
  * Or it could be a value between `0.0625` and `0.9375` (basically aligning with the 16x16 textures) and round to the nearest `1/16`th (in-range) -- I'd propose naming it `setHeight()` instead
* `getHeightMode() Returns a value corresponding to the configured height mode (name this `getHeight()` instead if it's not predetermined modes)
* `getDistance()` would return the distance (in blocks/metres) of the detected block or entity. Returns `-1` if nothing is detected in range.

If the sensor does not detect something as it's out of range, then it should probably return a value of `-1`.

**Comment by Seniorendi** [comment]
For what would you need the height? For something like slabs?

**Comment by Peekofwar** [comment]
> For what would you need the height? For something like slabs?

Yeah, just some way to adjust for low-obstacles like slabs could be useful. Not really necessary, but would be a nice thing to have if you need it.

**Comment by Peekofwar** [comment]
Oh, another suggestion to add to this: Could have an event fire when it detects something, or maybe an option to enable events.
```lua
{
   "laser_tripped",
   "laser_sensor_7",
   4.3,
}
```
Just a thought.

**Comment by SirEndii** [comment]
https://user-images.githubusercontent.com/67484093/227661737-9d15c58e-9b0c-4e61-8629-de6a093ecb15.mp4

A sneak peek. Almost ready

**Comment by Peekofwar** [comment]
Can it be vertically oriented, or is it strictly horizontal?

**Comment by SirEndii** [comment]
It can be oriented in any direction
